std/json.d

```diff
@@ -243,32 +243,68 @@ struct JSONValue
     /// Value getter/setter for $(D JSON_TYPE.OBJECT).
     /// Throws: $(D JSONException) for read access if $(D type) is not
     /// $(D JSON_TYPE.OBJECT).
-    @property ref inout(JSONValue[string]) object() inout pure
+    /* Note: this is @system because of the following pattern:
+       ---
+       auto a = &(json.object());
+       json.uinteger = 0;        // overwrite AA pointer
+       (*a)["hello"] = "world";  // segmentation fault
+       ---
+     */
+    @property ref inout(JSONValue[string]) object() inout pure @system
     {
         enforce!JSONException(type == JSON_TYPE.OBJECT,
                                 "JSONValue is not an object");
         return store.object;
     }
     /// ditto
-    @property JSONValue[string] object(JSONValue[string] v) pure nothrow @nogc
+    @property JSONValue[string] object(JSONValue[string] v) pure nothrow @nogc @safe
     {
         assign(v);
+        return v;
+    }
+
+    /// Value getter for $(D JSON_TYPE.OBJECT).
+    /// Unlike $(D object), this retrieves the object by value and can be used in @safe code.
+    /// Throws: $(D JSONException) for read access if $(D type) is not
+    /// $(D JSON_TYPE.OBJECT).
+    @property inout(JSONValue[string]) objectNoRef() inout pure @trusted
```

Comment:
The documentation for this should explain why the ref-returning `object` is there in the first place - i.e. this code:
```d
string[string] aa;
JSONValue v = aa;
v.objectNoRef["foo"] = JSONValue("bar");
assert("foo" !in v.object); // Because v.objectNoRef was null, a new AA was lazily initialized at the call-site without updating the still-null reference in v
```
Our crappy lazy initialization for reference types setup strikes again.